src/app/components/mno-apps/mno-app-listing.coffee

```diff
@@ -84,11 +84,12 @@ angular.module 'mnoEnterpriseAngular'
             response = response.plain()
 
             vm.categories = response.categories
-            vm.publicCategories = _.map(response.categories, (c) -> {label: c, active: ''})
             vm.apps = response.apps
-            vm.filteredApps = vm.apps
             if vm.publicPage
               vm.apps = _.filter(vm.apps, (app) -> _.includes(MnoeConfig.publicApplications(), app.nid))
+              categoriesWithProduct = _.uniq(_.flatten(_.map(vm.apps, (app) -> app.categories)))
```

Comment:
You can use the `_.property` callback shorthand:
```coffeescript
_.map(vm.apps, 'categories')
```

Reply:
I realise all this filtering for public apps is done in the frontend.
Would it be better to move it to the backend?

We already have the categories filtering logic:
* https://github.com/maestrano/mno-enterprise/blob/4.0/api/lib/mno_enterprise/concerns/controllers/jpi/v1/marketplace_controller.rb#L36
* https://github.com/maestrano/mno-enterprise/blob/4.0/core/app/models/mno_enterprise/app.rb#L16

Reply:
It is certainly a reasonable option. It'd remove the filtering in three locations (landing page, and the two app listing components).

Reply:
r.e. the property callback.  I wasn't aware of the shorthand. Thanks for pointing it out.